File: secondSeminar/src/main/java/org/sopt/secondSeminar/dto/response/MemberGetResponse.java
package org.sopt.secondSeminar.dto.response;

import lombok.AllArgsConstructor;
import lombok.Data;
import org.sopt.secondSeminar.domain.Member;
import org.sopt.secondSeminar.domain.SOPT;

@AllArgsConstructor
@Data
public class MemberGetResponse {
    private String name;
    private String nickname;
    private int age;
    private SOPT sopt;

    public static MemberGetResponse of(Member member) {
        return new MemberGetResponse(
                member.getName(),
                member.getNickname(),
                member.getAge(),
                member.getSopt()
        );
    }
}

File: secondKotlin/src/main/kotlin/org/sopt/secondKotlin/dto/response/MemberGetResponse.kt
package org.sopt.secondKotlin.dto.response

import org.sopt.secondKotlin.domain.Member
import org.sopt.secondKotlin.domain.SOPT

class MemberGetResponse(
    val name: String,
    val nickname: String,
    val age: Int,
    val sopt: SOPT
) {
    companion object {
        fun of(
            member: Member
        ): MemberGetResponse {
            return MemberGetResponse(
                member.name,
                member.nickname,
                member.age,
                member.SOPT
            )
        }
    }
}